onebusaway-android/src/main/java/org/onebusaway/android/nav/NavigationServiceProvider.java

```diff
@@ -706,4 +716,113 @@ private void silence(long duration, int queueFlag) {
             mTTS.playSilence(duration, queueFlag, null);
         }
     }
+
+    public void getUserFeedback() {
+        //TODO - Log "Yes" or "No" including plaintext feedback using Firebase Analytics
+
+        Application app = Application.get();
+        NotificationCompat.Builder mBuilder;
+
+        String message = Application.get().getString(R.string.feedback_notify_dialog_msg);
+        mFirstFeedback = Application.getPrefs().getBoolean(FIRST_FEEDBACK, true);
+
+        // Create delete intent to set flag for snackbar creation next time the app is opened.
+        Intent rIntent = new Intent(app.getApplicationContext(), FeedbackReceiver.class);
+        rIntent.putExtra(FeedbackReceiver.NOTIFICATION_ID, NOTIFICATION_ID + 1);
+
+        if ((mFirstFeedback) || (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) || (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P)) {
+
+            Intent fdIntent = new Intent(app.getApplicationContext(), FeedbackActivity.class);
+            fdIntent.setAction(REPLY_ACTION);
+            fdIntent.putExtra("CallingAction", "no");
```

Comment:
Per further discussion, let's leave these as extras and just rename them from actions to "response", or something similar.

Reply:
We have renamed `"CallingAction"` to `"response"` so that we can clearly distinguish between the REPLY and DISMISS actions and the user feedback.